src/orion/client/runner.py

```diff
@@ -0,0 +1,339 @@
+            # already no need to worry about it
+            # NB: suggest reserve the trial already
+            new_trials = self._suggest_trials(n_trial)
+            log.debug(f"Sampled {len(new_trials)} new configs")
+            return new_trials
+
+        return []
+
+    def scatter(self, new_trials):
+        """Schedule new trials to be computed"""
+        new_futures = []
+        for trial in new_trials:
+            future = self.client.executor.submit(
+                _optimize, trial, self.fct, self.trial_arg, **self.kwargs
+            )
+            self.pending_trials[future] = trial
+            new_futures.append(future)
+
+        self.futures.extend(new_futures)
+        log.debug("Scheduled new trials")
+
+    def gather(self):
+        """Gather the results from each worker asynchronously"""
+        results = []
+        try:
+            results = self.client.executor.async_get(
+                self.futures, timeout=self.gather_timeout
+            )
+        except KeyboardInterrupt:
+            self._release_all()
+            raise
+
+        to_be_raised = None
+        log.debug(f"Gathered new results {len(results)}")
+
+        # register the results
+        for result in results:
+            trial = self.pending_trials.pop(result.future)
+
+            if isinstance(result, AsyncResult):
+                try:
+                    # NB: observe release the trial already
+                    self.client.observe(trial, result.value)
+                    self.trials += 1
+                except InvalidResult as exception:
+                    # stop the optimization process if we received `InvalidResult`
+                    # as all the trials are assumed to be returning those
+                    to_be_raised = exception
+                    self.client.release(trial, status="broken")
+
+            if isinstance(result, AsyncException):
+                if (
+                    isinstance(result.exception, ExecutionError)
+                    and result.exception.return_code == self.interrupt_signal_code
+                ):
+                    to_be_raised = KeyboardInterrupt()
+                    self.client.release(trial, status="interrupted")
+                    continue
+
+                # Regular exception, might be caused by the choosen hyperparameters
+                # themselves rather than the code in particular (like Out of Memory error
+                # for big batch sizes)
+                exception = result.exception
+                self.worker_broken_trials += 1
+                self.client.release(trial, status="broken")
+
+                if self.on_error is None or self.on_error(
+                    self, trial, exception, self.worker_broken_trials
+                ):
+                    log.error(result.traceback)
+
+                else:
+                    log.error(str(exception))
+                    log.debug(result.traceback)
```

Comment:
Codecov reports these line as not tested but it should be covered in [test_runner.py::TestWorkon::test_workon_on_error_ignore](https://github.com/Epistimio/orion/pull/684/files#diff-f06b94773052cdf9a38bfdb3ea74cb852af76b54d9c1acf6841cd98db127d04cR627) and [test_runner.py::TestWorkon::test_workon_on_error_raise](https://github.com/Epistimio/orion/pull/684/files#diff-f06b94773052cdf9a38bfdb3ea74cb852af76b54d9c1acf6841cd98db127d04cR681).

Reply:
so ?

Reply:
So nothing to do, just writing it down here as a reference so we understand why we did not try to address these reports from codecov.